### PR TITLE
Add rule sssd_run_as_sssd_user for SSSD to drop root privileges.

### PR DESCRIPTION
**Pull request by adelton**
#### Description:

- Add rule sssd_run_as_sssd_user for SSSD to drop root privileges.

#### Rationale:

- Running SSSD as non-root decreases the chance of bad things happening.

This is an initial stab at the rule. Missing are the `identifiers` and `references` in `rule.yml` and check for the SSSD service being enabled. I need some guidance on those.

### CODE REVIEW

File: linux_os/guide/services/sssd/sssd_run_as_sssd_user/rule.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    To minimize privileges of SSSD processes, they are configured to
+    run as non-root user.
+
+severity: medium
+
+identifiers:
+    cce@rhel8: 12345-5
+
+references:
+    ospp: FIA_AFL.1-FIXME-and-identifiers
```
Comment (yuumasato) [inline linux_os/guide/services/sssd/sssd_run_as_sssd_user/rule.yml]: Regarding OSPP, I could not identify to which requirement this rule is related.
Closest thing was an Objective for the Operational Environment for the user,([OE.PROPER_USER](https://www.niap-ccevs.org/MMO/PP/-442-/#SecurityObjectivesTOEorEnvironment)), but it is not related to service users, though.

But I found requirements that support this rule, like:

- [FPT_ACF_EXT.1.1](https://www.niap-ccevs.org/MMO/PP/-442-/#FPT_ACF_EXT.1.1)
- [FPT_ACF_EXT.1.1](https://www.niap-ccevs.org/MMO/PP/-442-/#FDP_ACF_EXT.1.1)

@stevegrubb Can you give us a hint?

**Comment by adelton** [comment]
> that it should be evaluated when `sssd.service` is installed in the system

Well, better applicability check would be to test if the `sssd.service` is enable, wouldn't it? Do we have a way to do that easily?

**Comment by yuumasato** [comment]
> > that it should be evaluated when `sssd.service` is installed in the system
> 
> Well, better applicability check would be to test if the `sssd.service` is enable, wouldn't it? 

One could argue that if the service is installed, there is possibility to configure it. Independent of its active or enabled state.

>Do we have a way to do that easily?

Yes, instead of checking for package installed with `rpminfo`, checking for the service status with `systemdunitproperty`.

**Comment by adelton** [comment]
Updated `identifiers` and `references`, rebased on master -> a43134a0cf91850b24ea0e8b78b25dda29fdef3f.

**Comment by adelton** [comment]
> As we briefly discussed, only issue with this rule is WRT to applicability of the rule, that it should be evaluated when `sssd.service` is installed in the system. To address this, something like `platform: sssd` in the `rule.yml` could be introduced for SSSD rules.
> 
> But I don't consider this issue as blocking this PR.
> See, #4241 to have an idea how it is like.

I'm now adding it to this pull request -> a7a6ab3c5295f5042f7afe5bf622e80b7fb79509.

Ideally we would check for the service to be also enabled but let's leave this for the future. Subsequent change (likely after this PR is merged) would then get rid of the `service_sssd_disabled` criterions in other SSSD OVALs.

**Comment by adelton** [comment]
I've now updated the first commit to remove the proposed CEE from shared/references/cce-redhat-avail.txt, per recommendation in https://github.com/ComplianceAsCode/content/pull/4445#issuecomment-505304858.

File: linux_os/guide/services/sssd/sssd_run_as_sssd_user/rule.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    run as non-root user.
+
+severity: medium
+
+identifiers:
+    cce@rhel8: 81062-2
+
+references:
+    ospp: FPT_ACF_EXT.1.1
+    ospp: FDP_ACF_EXT.1.1
```
Comment (yuumasato) [inline linux_os/guide/services/sssd/sssd_run_as_sssd_user/rule.yml]: So it seems that the correct reference should be `FMT_SMF_EXT.1`.

File: ssg/constants.py [hunk 3]
```diff
@@ -383,6 +383,7 @@
     "shadow-utils": "cpe:/a:shadow-utils",
     "systemd": "cpe:/a:systemd",
     "yum": "cpe:/a:yum",
+    "sssd": "cpe:/a:sssd",
```
Comment (yuumasato) [inline ssg/constants.py]: nitpick: Please keep the CPEs ordered alphabetically after `machine` and `container` platform CPEs.

File: linux_os/guide/services/sssd/sssd_run_as_sssd_user/rule.yml (hunk 2 above)
Comment (adelton) [reply linux_os/guide/services/sssd/sssd_run_as_sssd_user/rule.yml]: Fixed in 57abcf27fdf6d46c0251e0fc4a916a5d163a0662.

File: ssg/constants.py (hunk 3 above)
Comment (adelton) [reply ssg/constants.py]: Fixed in 8fb9b8c2cfeb7659952fab86a69739ff9f05edd6.

**Comment by yuumasato** [comment]
@openscap-ci test this please

**Comment by yuumasato** [comment]
Something strange is going on with Jenkins...

**Comment by yuumasato** [comment]
Thank you for the rule.

**Pull request merged**